tests/components/bayesian/test_binary_sensor.py

```diff
@@ -18,6 +19,65 @@ def teardown_method(self, method):
         """Stop everything that was started."""
         self.hass.stop()
 
+    def test_load_values_when_added_to_hass(self):
+        """Test that sensor initializes with observations of relevant entities."""
+
+        config = {
+            "binary_sensor": {
+                "name": "Test_Binary",
+                "platform": "bayesian",
+                "observations": [
+                    {
+                        "platform": "state",
+                        "entity_id": "sensor.test_monitored",
+                        "to_state": "off",
+                        "prob_given_true": 0.8,
+                        "prob_given_false": 0.4,
+                    }
+                ],
+                "prior": 0.2,
+                "probability_threshold": 0.32,
+            }
+        }
+
+        self.hass.states.set("sensor.test_monitored", "off")
+        self.hass.block_till_done()
+
+        assert setup_component(self.hass, "binary_sensor", config)
+
+        state = self.hass.states.get("binary_sensor.test_binary")
+        assert state.attributes.get("observations")[0]["prob_given_true"] == 0.8
+        assert state.attributes.get("observations")[0]["prob_given_false"] == 0.4
+
+    def test_unknown_state_does_not_influence_probability(self):
+        """Test that an unknown state does not change the output probability."""
+
+        config = {
+            "binary_sensor": {
+                "name": "Test_Binary",
+                "platform": "bayesian",
+                "observations": [
+                    {
+                        "platform": "state",
+                        "entity_id": "sensor.test_monitored",
+                        "to_state": "off",
+                        "prob_given_true": 0.8,
+                        "prob_given_false": 0.4,
+                    }
+                ],
+                "prior": 0.2,
+                "probability_threshold": 0.32,
+            }
+        }
+
+        self.hass.states.set("sensor.test_monitored", STATE_UNKNOWN)
+        self.hass.block_till_done()
+
+        assert setup_component(self.hass, "binary_sensor", config)
+
+        state = self.hass.states.get("binary_sensor.test_binary")
+        assert state.attributes.get("observations") == [None]
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5279916)